lib/minefield.rb

```diff
@@ -0,0 +1,198 @@
+require_relative 'cell'
+
+class Minefield
+
+  attr_reader :size, :field
+
+  def initialize(size = 10)
+    set_field_size(size)
+    create_field_with_attributes
+  end
+
+  def render
+    field.each do |row|
+      row.each { |cell| print cell }
+      print "\n"
+    end
+    nil
+  end
+
+  def take_turn(turn)
+    row = turn[:row]
+    column = turn[:column]
+    action = turn[:action]
+    take_action(row, column, action)
+  end
+
+  # GAME ENDING SCENARIOS
+
+  def lost?
+    field.flatten.any? { |cell| cell.exploded? }
+  end
+
+  def blow_up_board
+    field.flatten.each do |cell|
+      cell.clear if cell.mine
```

Comment:
In the demo I really enjoyed this feature, seeing the mines blow up upon loss. Good job!